gtest/src/manager.rs

```diff
@@ -948,6 +948,40 @@ impl ExtManager {
 
         core_processor::handle_journal(journal, self);
     }
+
+    fn remove_reservation(&mut self, id: ProgramId, reservation: ReservationId) -> Option<()> {
+        let res = self.adjust_genuine_program(id, |genuine_program| {
+            let genuine_program = genuine_program.expect("no genuine program was found");
+            genuine_program
+                .gas_reservation_map
+                .remove(&reservation)
+                .map(|_| ())
+                .or_else(|| {
+                    log::error!("Tried to remove non-existent reservation");
+
+                    None
+                })
+        });
+
+        res.map(|_| {
+            self.gas_tree
+                .consume(reservation)
+                .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+        })
+    }
+
+    pub(crate) fn adjust_genuine_program<R, F: FnOnce(Option<&mut GenuineProgram>) -> R>(
```

Comment:
```suggestion
    pub(crate) fn adjust_genuine_program<R, F: FnOnce(&mut GenuineProgram) -> R>(
```
Every usage unwraps `Option`. Consider to unwrap it in function itself

Reply:
well than you have to return from the `adjust_genuine_program` the `Option` then, as the option inside the method won't be unwraped, as it's wrong.

What's more, with such an API you will have to handle 2 results - 1) for the `op` one and for the `adjust_genuine_program`

Reply:
done